config/config.go

```diff
@@ -305,6 +305,12 @@ func (c *configStruct) ChooseEndpoint(overridingEndpointAliasOrURL string) strin
 		}
 
 		ep := normalizeEndpoint(overridingEndpointAliasOrURL)
+
+		// overwrite c.SelectedEndpoint to make the override accessible for
+		// other calls to the library
+		// Don't use c.SelectEndpoint() as it writes to the config file
+		c.SelectedEndpoint = ep
```

Comment:
@yulianedyalkova could you take a look at this?

I'm unsure if this needs some mutex stuff like in SelectEndpoint

https://github.com/giantswarm/gscliauth/blob/master/config/config.go#L26